pkg/analysis/numericbounds/analyzer.go

```diff
@@ -0,0 +1,216 @@
+	minMarkers, maxMarkers := getMarkerNames(isSlice)
+
+	// Get minimum and maximum marker values
+	minimum, minErr := getMarkerNumericValue(fieldMarkers, minMarkers)
+	maximum, maxErr := getMarkerNumericValue(fieldMarkers, maxMarkers)
+
+	// Check if markers are missing
+	minMissing := errors.Is(minErr, errMarkerMissingValue)
+	maxMissing := errors.Is(maxErr, errMarkerMissingValue)
+
+	// Report any invalid marker values (e.g., non-numeric values)
+	if minErr != nil && !minMissing {
+		pass.Reportf(field.Pos(), "%s has an invalid minimum marker: %v", prefix, minErr)
+	}
+
+	if maxErr != nil && !maxMissing {
+		pass.Reportf(field.Pos(), "%s has an invalid maximum marker: %v", prefix, maxErr)
+	}
+
+	// Report if markers are missing
+	if minMissing {
+		pass.Reportf(field.Pos(), "%s is missing minimum bounds validation marker", prefix)
+	}
+
+	if maxMissing {
+		pass.Reportf(field.Pos(), "%s is missing maximum bounds validation marker", prefix)
+	}
+
+	// If any markers are missing or invalid, don't continue with bounds checks
+	if minErr != nil || maxErr != nil {
+		return
+	}
+
+	// Validate bounds are within the type's valid range
+	checkBoundsWithinTypeRange(pass, field, prefix, ident.Name, minimum, maximum)
+}
+
+// getMarkerNames returns the appropriate minimum and maximum marker names
+// based on whether the field is a slice.
+// Returns both kubebuilder and k8s declarative validation markers.
+func getMarkerNames(isSlice bool) (minMarkers, maxMarkers []string) {
+	if isSlice {
+		return []string{markers.KubebuilderItemsMinimumMarker}, []string{markers.KubebuilderItemsMaximumMarker}
+	}
+
+	return []string{markers.KubebuilderMinimumMarker, markers.K8sMinimumMarker}, []string{markers.KubebuilderMaximumMarker, markers.K8sMaximumMarker}
+}
+
+// getMarkerNumericValue extracts the numeric value from the first instance of any of the given marker names.
+// Checks multiple marker names to support both kubebuilder and k8s declarative validation markers.
+func getMarkerNumericValue(markerSet markershelper.MarkerSet, markerNames []string) (float64, error) {
+	for _, markerName := range markerNames {
+		markerList := markerSet.Get(markerName)
+		if len(markerList) == 0 {
+			continue
+		}
+
+		// Use the exported utils function to parse the marker value
+		value, err := utils.GetMarkerNumericValue[float64](markerList[0])
+		if err != nil {
+			if errors.Is(err, errMarkerMissingValue) {
+				continue
+			}
+
+			return 0, fmt.Errorf("error getting marker value: %w", err)
+		}
+
+		return value, nil
+	}
+
+	return 0, errMarkerMissingValue
+}
+
+// checkBoundsWithinTypeRange validates that the bounds are within the valid range for the type.
+// For int64, enforces JavaScript-safe bounds as per Kubernetes API conventions to ensure
+// compatibility with JavaScript clients.
+//
+//nolint:cyclop
+func checkBoundsWithinTypeRange(pass *analysis.Pass, field *ast.Field, prefix, typeName string, minimum, maximum float64) {
```

Comment:
If you were to use generics for these bounds checks, you might be able to avoid int to float to int conversions on the integer bounds checks

I suspect a bounds checking function that takes a boundary, a message, an upper and lower boundary could be leveraged here